cppcache/src/PdxInstanceImpl.hpp

```diff
@@ -211,35 +210,36 @@ class PdxInstanceImpl : public WritablePdxInstance {
 
   void operator=(const PdxInstanceImpl& other) = delete;
 
-  std::shared_ptr<PdxType> getPdxType() const;
+  std::shared_ptr<PdxType> getPdxType(Pool* pool) const;
 
-  void updatePdxStream(uint8_t* newPdxStream, int len);
+  int32_t getTypeId() const;
+
+  void updatePdxStream(std::vector<uint8_t> stream);
+
+ protected:
+  const std::vector<uint8_t>& getPdxStream() const;
 
  private:
-  std::vector<uint8_t> m_buffer;
-  int m_typeId;
+  mutable std::vector<uint8_t> m_buffer;
+  mutable int32_t m_typeId;
```

Comment:
This is by convention member variables that are lazily initializated are supposed to be mutable. The reason is because once initialized they are supposed to be immutable, but sadly there is no support from the language for lazy initialization, so that's the way it's usually done.